src/index.ts

```diff
@@ -515,6 +545,26 @@ export function create(rawOptions: CreateOptions = {}): Service {
     ...(rawOptions.require || []),
   ];
 
+  // Experimental REPL await is not compatible targets lower than ES2018
+  const targetSupportsTla = config.options.target! >= ScriptTarget.ES2018;
+  if (options.experimentalReplAwait === true && !targetSupportsTla) {
+    throw new Error(
+      'Experimental REPL await is not compatible with targets lower than ES2018'
+    );
+  }
+  // Top-level await was added in TS 3.8
+  const tsVersionSupportsTla = versionGte(ts.version, '3.8.0');
+  if (options.experimentalReplAwait === true && !tsVersionSupportsTla) {
+    throw new Error(
+      'Experimental REPL await is not compatible with TypeScript versions older than 3.8'
+    );
+  }
+
+  const shouldReplAwait =
+    options.experimentalReplAwait !== false &&
```

Comment:
Couldn't the conditional be just `options.experimentalReplAwait !== false`? any other invalid combination already thrown an error.

Reply:
Not quite, because `options.experimentalReplAwait` can be one of 3 values: `true`, `false`, or `undefined`.  `true` raises the errors, but `undefined` does not.

Reply:
I see that `options.experimentalReplAwait` is conditionally set, and that could lead to confusions on why it's not working on lower targets, for example, running `node ./dist/bin.js`:

```typescript
> await 1;
<repl>.ts:1:1 - error TS1375: 'await' expressions are only allowed at the top level of a file when that file is a module, but this file has no imports or exports. Consider adding an empty 'export {}' to make this file a module.

1 await 1;
  ~~~~~
<repl>.ts:1:1 - error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext' or 'system', and the 'target' option is set to 'es2017' or higher.

1 await 1;
  ~~~~~

undefined
>
```

the default value was set automatically to `false` (which is not the node default), but there's no warning given that the issue is actually setting a higher target, I think we should print a warning/hint if a relevant diagnostic code is part of the error thrown in `_eval` - `service.compile`, pointing the user to set `es2018`+

Reply:
Agreed; great idea.  In my opinion, there are a bunch of places in ts-node where we need to be proactively showing helpful error or warning messages, similar to this.

The error can include the current target if less than es2018, and can include the current TS version if less than 3.8.  Something like "you are using ts 3.7; must use 3.8 or higher"

Perhaps it can also recommend `ts-node --showConfig` and `ts-node -vv` to debug config issues.